### PR TITLE
`azurerm_kusto_script` - Lock kusto cluster so multiple scripts can be applied

### PR DESCRIPTION
**Pull request by mojanas**
Fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/16471

```bash
$ make acctests SERVICE='kusto' TESTARGS='-run=TestAccKustoScript_multiple' TESTTIMEOUT='30m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/kusto -run=TestAccKustoScript_multiple -timeout 30m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccKustoScript_multiple
=== PAUSE TestAccKustoScript_multiple
=== CONT  TestAccKustoScript_multiple
--- PASS: TestAccKustoScript_multiple (1412.46s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto 1415.445s
```

Based off of https://github.com/hashicorp/terraform-provider-azurerm/pull/16649
The kusto scripts modify the underlying cluster configuration during apply, so we would want to lock the cluster instead of the database.

### CODE REVIEW

**Comment by mojanas** [comment]
Tagging @mbfrahry and @sinbai for visibility

File: internal/services/kusto/kusto_database_script_resource.go [hunk 1]
```diff
@@ -98,8 +98,8 @@ func resourceKustoDatabaseScriptCreateUpdate(d *pluginsdk.ResourceData, meta int
 		}
 	}
 
-	locks.ByID(databaseId.ID())
-	defer locks.UnlockByID(databaseId.ID())
+	locks.ByID(id.ClusterName)
```
Comment (mbfrahry) [inline internal/services/kusto/kusto_database_script_resource.go:101]: This needs to be the actual cluster id and not just the name. We can get that id with 
```	
clusterId := parse.NewClusterID(databaseId.SubscriptionId, databaseId.ResourceGroup, databaseId.ClusterName)
locks.ByID(clusterId.ClusterName)
defer locks.UnlockByID(clusterId.ClusterName)
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @mojanas, this is nearly there but we need to lock the actual cluster id and not just the cluster name.

File: internal/services/kusto/kusto_database_script_resource.go (hunk 1 above)
Comment (mojanas) [reply internal/services/kusto/kusto_database_script_resource.go]: Updated

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**